src/services/dashboards/dashboard-detail/modules/DashboardNameEditModal.vue

```diff
@@ -89,28 +92,38 @@ export default defineComponent<Props>({
             proxyVisible: props.visible,
         });
 
+        const isProjectDashboard = Boolean(props.dashboardId?.startsWith('project'));
+
         const updateDashboard = async () => {
             try {
-                if (props.dashboardId.startsWith('user')) {
-                    // TODO:: connect custom-dashboard update api
-                    // await SpaceConnector.client.costAnalysis.userDashboard.update({
-                    //     user_dashboard_id: props.dashboardId,
-                    //     name: name.value,
-                    // });
+                if (isProjectDashboard) {
+                    await SpaceConnector.clientV2.dashboard.projectDashboard.update({
+                        project_dashboard_id: props.dashboardId,
+                        name: name.value,
+                    });
                 } else {
-                    // await SpaceConnector.client.costAnalysis.publicDashboard.update({
-                    //     public_dashboard_id: props.dashboardId,
-                    //     name: name.value,
-                    // });
+                    await SpaceConnector.clientV2.dashboard.domainDashboard.update({
+                        domain_dashboard_id: props.dashboardId,
+                        name: name.value,
+                    });
                 }
             } catch (e) {
-                ErrorHandler.handleError(e);
+                // song-lang
+                ErrorHandler.handleRequestError(e, 'Failed to update dashboard name');
+            }
+        };
+
+        const loadDashboard = async () => {
+            if (isProjectDashboard) {
+                await store.dispatch('dashboard/loadProjectDashboard');
+            } else {
+                await store.dispatch('dashboard/loadDomainDashboard');
             }
         };
 
         const handleConfirm = () => {
             updateDashboard();
```

Comment:
suggestion: I think that this needs to be async.

```suggestion
        const handleConfirm = async () => {
            await updateDashboard();
```

Reply:
Oh, I forgot it. Thanks a lot